### PR TITLE
Upgrade Font Awesome to 5 and decrease contact icon jump

### PR DESCRIPTION
**Pull request by pawelngei**
Upgrading allows us to have more social media buttons, including Mastodon I use at https://alxd.org/

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -121,7 +123,9 @@ <h1><a href="{{ SITEURL }}">{{ SITETITLE }}</a></h1>
 
       <ul class="social">
         {% for name, link in SOCIAL %}
-        <li><a class="sc-{{ name }}" href="{{ link }}" target="_blank"><i class="fa fa-{{ name }}"></i></a></li>
+          <li><a class="sc-{{ name }}" href="{{ link }}" target="_blank">
+            <i class="fa{% if name in ['envelope-open', 'rss'] %}s{% else %}b{% endif %} fa-{{ name }}"></i>
```
Comment (pawelngei) [inline templates/base.html]: If you want I can split this into more verbose else/if with several names. The problem is caused by the introduction of font "families" in Font Awesome 5: `envelope-open` and `rss` need `fas`, and all the other social media `fab` class.

File: static/stylesheet/style.less [hunk 2]
```diff
@@ -430,7 +430,7 @@ ul.social {
     background-color: @github-bg-color;
   }
 
-  a.sc-envelope-o {
+  a.sc-envelope-open {
```
Comment (alexandrevicenzi) [inline static/stylesheet/style.less]: not sure if this is going to break something

File: templates/base.html (hunk 1 above)
Comment (alexandrevicenzi) [reply templates/base.html]: my suggestion it just to use `fas` and `fab` inside if else, it's more readable.

File: static/stylesheet/style.less (hunk 2 above)
Comment (pawelngei) [reply static/stylesheet/style.less]: In the new `font-awesome` there is no `envelope-o`, but `envelope-open`, so the migration would require changing the name. I can't find any place which would be broken by this.

**Comment by pawelngei** [comment]
Updated the `base.html` as per request, but I didn't want to touch the `docs` for now. Should I remake the whole example blog to use the FA-5 as well?

I also submitted a separate #160 which could be useful outside of the icon issue.

File: templates/base.html [hunk 3]
```diff
@@ -124,7 +124,13 @@ <h1><a href="{{ SITEURL }}">{{ SITETITLE }}</a></h1>
       <ul class="social">
         {% for name, link in SOCIAL %}
           <li><a class="sc-{{ name }}" href="{{ link }}" target="_blank">
-            <i class="fa{% if name in ['envelope-open', 'rss'] %}s{% else %}b{% endif %} fa-{{ name }}"></i>
+            <i class="{% if name in ['envelope-open', 'rss'] %}
```
Comment (alexandrevicenzi) [inline templates/base.html]: one line is better, otherwise it will output to many extra spaces

File: static/stylesheet/style.less (hunk 2 above)
Comment (alexandrevicenzi) [reply static/stylesheet/style.less]: it's not in Flex source, it's in pelicancong.py, when you choose to show your email as an icon.

**Comment by pawelngei** [comment]
Thank you for the comments! I will not have access to the internet for the next week, I'll fix it afterwards!

**Comment by pawelngei** [comment]
I collapsed the long else/if and replaced `envelope-o`/`envelope-open` with just `envelope`. Everything seems to be working both in the docs and on my own blog.

**Comment by alexandrevicenzi** [comment]
@pawelngei Thanks.

**Pull request merged**